src/fits.cc

```diff
@@ -1331,6 +1331,25 @@ bool Fits::checkImageType() {
     }
 }
 
+std::string Fits::getImageDType() {
+    int bitpix = 0;
+    fits_get_img_equivtype(reinterpret_cast<fitsfile *>(fptr), &bitpix, &status);
+    if (behavior & AUTO_CHECK) LSST_FITS_CHECK_STATUS(*this, "Getting image type");
+    if (bitpix < 0) {
+        return "float" + std::to_string(-bitpix);
+    }
+    switch (bitpix) {
+        case BYTE_IMG: return "uint8";
+        case SBYTE_IMG: return "int8";
+        case SHORT_IMG: return "int16";
+        case USHORT_IMG: return "uint16";
+        case LONG_IMG: return "int32";
+        case ULONG_IMG: return "uint32";
+        case LONGLONG_IMG: return "int64";
+    }
+    return "";
```

Comment:
I was about to say that maybe it should be an uncontrolled panic, because that would mean a definite logic bug, but I suppose when you're reading from disk you need to have an open mind about what you could get.  I've turned it into an exception.